lib/config/index.js

```diff
@@ -96,11 +97,21 @@ config.isGitLabEnable = config.gitlab.clientID && config.gitlab.clientSecret
 config.isMattermostEnable = config.mattermost.clientID && config.mattermost.clientSecret
 config.isLDAPEnable = config.ldap.url
 config.isSAMLEnable = config.saml.idpSsoUrl
-config.isPDFExportEnable = config.allowpdfexport
+config.isPDFExportEnable = config.allowPDFExport
 
 // merge legacy values
-if (config.imageUploadType && !config.imageuploadtype) {
-  config.imageuploadtype = config.imageUploadType
+let keys = Object.keys(config)
+for (let i = keys.length; i--;) {
+  const uppercase = /[A-Z]/
+  // if the config contains uppercase letters
+  // and a lowercase version of this setting exists
+  // and the original, new, config with uppercase is not set
+  // in this case we set the new config using the old key.
+  if (uppercase.test(keys[i]) &&
+    config[keys[i].toLowerCase()] &&
```

Comment:
Through the following addition, we'll be using `keys[i].toLowerCase()` quite a bit, so it's probably a good idea to `let lowercaseCandidate = keys[i].toLowerCase()` this.

Reply:
:+1: